### ISSUE TITLE
Issues with setting up ingress across DEV and SYST clusters. AGIC 1.2, AKS 1.16.10 and Gateway Standard v2

### ISSUE THREAD

**Issue by sudheeranguluri** [Azure/application-gateway-kubernetes-ingress#979]
**A clear and concise description of what the bug is.**
We have two clusters (DEV and SYST)  and one AG. Installed AGIC Pod in each cluster and setup prohibited targets to not to disturb the setup in other cluster. Example of Ingress prohibited targets

**DEV**
apiVersion: appgw.ingress.k8s.io/v1
kind: AzureIngressProhibitedTarget
metadata:
  namespace: default
  name: prohibit-syst-target
spec:
    paths:
    - /oneapi_syst/*
    - /twoapi_syst/*

**SYST**
apiVersion: appgw.ingress.k8s.io/v1
kind: AzureIngressProhibitedTarget
metadata:
  namespace: default
  name: prohibit-dev-target
spec:
    paths:
    - /oneapi_dev/*
    - /twoapi_dev/*

**To Reproduce**
Deploy dev apis on DEV cluster and AGIC creates routes successfully in Gateway. Deploy syst apis on SYST cluster and AGIC is failing to apply new changes to Gateway.

* Output of `kubectl logs <ingress controller>. 
I0814 14:26:31.140178       1 context.go:360] AzureIngressProhibitedTargets: default/prohibit-dev-target,default/prohibit-uat-target,default/prohibit-syst-target-to-be-deleted,default/prohibit-default-dev-target
I0814 14:26:31.140229       1 mutate_app_gateway.go:83] [brownfield] Prohibited targets: {"Path":"/oneapi_dev/*"}, {"Path":"/twoapi_dev/*"}
I0814 14:26:31.140240       1 prune.go:48] Original Ingress[0] Rules: [{Host: IngressRuleValue:{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/oneapi_syst/**,Backend:IngressBackend{ServiceName:oneapi-syst,ServicePort:{0 8080 },Resource:nil,},PathType:nil,},},}}}]
I0814 14:26:31.140423       1 targets.go:49] [brownfield] Target {"Path":"/oneapi_syst/**"} is not blacklisted
I0814 14:26:31.140442       1 prune.go:50] Sanitized Ingress[0] Rules: [{Host: IngressRuleValue:{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/oneapi_syst/**,Backend:IngressBackend{ServiceName:oneapi-syst,ServicePort:{0 8080 },Resource:nil,},PathType:nil,},},}}}]
I0814 14:26:31.140486       1 prune.go:48] Original Ingress[1] Rules: [{Host: IngressRuleValue:{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/twoapi_syst/**,Backend:IngressBackend{ServiceName:twoapi-syst,ServicePort:{0 80 },Resource:nil,},PathType:nil,},},}}}]
I0814 14:26:31.140539       1 targets.go:49] [brownfield] Target {"Path":"/twoapi_syst/**"} is not blacklisted
I0814 14:26:31.140550       1 prune.go:50] Sanitized Ingress[1] Rules: [{Host: IngressRuleValue:{HTTP:&HTTPIngressRuleValue{Paths:[]HTTPIngressPath{HTTPIngressPath{Path:/twoapi_syst/**,Backend:IngressBackend{ServiceName:twoapi-syst,ServicePort:{0 80 },Resource:nil,},PathType:nil,},},}}}]
I0814 14:26:31.140654       1 health_probes.go:55] Created default HTTP probe defaultprobe-Http
I0814 14:26:31.140666       1 health_probes.go:56] Created default HTTPS probe defaultprobe-Http
I0814 14:26:31.140677       1 ingress_rules.go:148] Found backend:one/oneapi-syst
I0814 14:26:31.140704       1 ingress_rules.go:148] Found backend:two/twoapi-syst
I0814 14:26:31.145809       1 health_probes.go:70] Created probe pb-two-twoapi-syst-80-twoapi-syst for ingress two/twoapi-syst at service two/twoapi-syst
I0814 14:26:31.145995       1 health_probes.go:70] Created probe pb-one-oneapi-syst-8080-oneapi-syst for ingress one/oneapi-syst at service one/oneapi-syst
I0814 14:26:31.146067       1 routing_rules.go:31] [brownfield] Rule to Targets map: map[rr-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:} {Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**}]]
I0814 14:26:31.146116       1 targets.go:49] [brownfield] Target {} is not blacklisted
I0814 14:26:31.146149       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.146166       1 routing_rules.go:39] [brownfield] Routing Rule rr-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.146186       1 pathmaps.go:29] [brownfield] PathMap to Targets map: map[url-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**}]]
I0814 14:26:31.146231       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.146244       1 pathmaps.go:37] [brownfield] Routing PathMap url-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.146259       1 http_settings.go:29] HTTP Setting bp-one-oneapi-dev-8080-8080-oneapi-dev is blacklisted
I0814 14:26:31.146278       1 http_settings.go:29] HTTP Setting bp-two-twoapi-dev-80-8080-twoapi-dev is blacklisted
I0814 14:26:31.146286       1 http_settings.go:29] HTTP Setting defaulthttpsetting is blacklisted
I0814 14:26:31.146298       1 health_probes.go:27] Probe defaultprobe-Http is blacklisted
I0814 14:26:31.146314       1 health_probes.go:31] Probe defaultprobe-Https is not blacklisted
I0814 14:26:31.146321       1 health_probes.go:27] Probe pb-one-oneapi-dev-8080-oneapi-dev is blacklisted
I0814 14:26:31.146327       1 health_probes.go:27] Probe pb-two-twoapi-dev-80-twoapi-dev is blacklisted
I0814 14:26:31.146343       1 health_probes.go:67] [brownfield] Probes AGIC created: defaultprobe-Http, defaultprobe-Https, pb-two-twoapi-syst-80-twoapi-syst, pb-one-oneapi-syst-8080-oneapi-syst
I0814 14:26:31.146361       1 health_probes.go:68] [brownfield] Existing Blacklisted Probes AGIC will retain: defaultprobe-Http, pb-one-oneapi-dev-8080-oneapi-dev, pb-two-twoapi-dev-80-twoapi-dev
I0814 14:26:31.146369       1 health_probes.go:69] [brownfield] Existing Probes AGIC will remove: 
I0814 14:26:31.146976       1 backendhttpsettings.go:190] Created backend http settings bp-two-twoapi-syst-80-8080-twoapi-syst for ingress two/twoapi-syst and service two/twoapi-syst
I0814 14:26:31.147036       1 backendhttpsettings.go:190] Created backend http settings bp-one-oneapi-syst-8080-8080-oneapi-syst for ingress one/oneapi-syst and service one/oneapi-syst
I0814 14:26:31.147068       1 routing_rules.go:31] [brownfield] Rule to Targets map: map[rr-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:} {Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**}]]
I0814 14:26:31.147112       1 targets.go:49] [brownfield] Target {} is not blacklisted
I0814 14:26:31.147135       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.147145       1 routing_rules.go:39] [brownfield] Routing Rule rr-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.147168       1 pathmaps.go:29] [brownfield] PathMap to Targets map: map[url-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**}]]
I0814 14:26:31.147209       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.147227       1 pathmaps.go:37] [brownfield] Routing PathMap url-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.147240       1 http_settings.go:29] HTTP Setting bp-one-oneapi-dev-8080-8080-oneapi-dev is blacklisted
I0814 14:26:31.147257       1 http_settings.go:29] HTTP Setting bp-two-twoapi-dev-80-8080-twoapi-dev is blacklisted
I0814 14:26:31.147264       1 http_settings.go:29] HTTP Setting defaulthttpsetting is blacklisted
I0814 14:26:31.147276       1 http_settings.go:68] [brownfield] HTTP Settings AGIC created: defaulthttpsetting, bp-two-twoapi-syst-80-8080-twoapi-syst, bp-one-oneapi-syst-8080-8080-oneapi-syst
I0814 14:26:31.147288       1 http_settings.go:69] [brownfield] Existing Blacklisted HTTP Settings AGIC will retain: bp-one-oneapi-dev-8080-8080-oneapi-dev, bp-two-twoapi-dev-80-8080-twoapi-dev, defaulthttpsetting
I0814 14:26:31.147295       1 http_settings.go:70] [brownfield] Existing HTTP Settings AGIC will remove: 
I0814 14:26:31.147323       1 backendaddresspools.go:37] Created default backend pool defaultaddresspool
I0814 14:26:31.147360       1 backendaddresspools.go:48] Created backend pool pool-two-twoapi-syst-80-bp-8080 for service two/twoapi-syst
I0814 14:26:31.147863       1 backendaddresspools.go:48] Created backend pool pool-one-oneapi-syst-8080-bp-8080 for service one/oneapi-syst
I0814 14:26:31.147897       1 routing_rules.go:31] [brownfield] Rule to Targets map: map[rr-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:} {Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**}]]
I0814 14:26:31.147928       1 targets.go:49] [brownfield] Target {} is not blacklisted
I0814 14:26:31.147965       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.147983       1 routing_rules.go:39] [brownfield] Routing Rule rr-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.148002       1 pathmaps.go:29] [brownfield] PathMap to Targets map: map[url-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**}]]
I0814 14:26:31.148038       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.148049       1 pathmaps.go:37] [brownfield] Routing PathMap url-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.148062       1 pools.go:28] [brownfield] Backend Address Pool defaultaddresspool is blacklisted
I0814 14:26:31.148078       1 pools.go:28] [brownfield] Backend Address Pool pool-one-oneapi-dev-8080-bp-8080 is blacklisted
I0814 14:26:31.148086       1 pools.go:28] [brownfield] Backend Address Pool pool-two-twoapi-dev-80-bp-8080 is blacklisted
I0814 14:26:31.148099       1 pools.go:67] [brownfield] Pools AGIC created: defaultaddresspool, pool-two-twoapi-syst-80-bp-8080, pool-one-oneapi-syst-8080-bp-8080
I0814 14:26:31.148116       1 pools.go:68] [brownfield] Existing Blacklisted Pools AGIC will retain: defaultaddresspool, pool-one-oneapi-dev-8080-bp-8080, pool-two-twoapi-dev-80-bp-8080
I0814 14:26:31.148123       1 pools.go:69] [brownfield] Existing Pools AGIC will remove: n/a
I0814 14:26:31.148142       1 frontend_listeners.go:121] Processing Rules for Ingress: one/oneapi-syst
I0814 14:26:31.148179       1 frontend_listeners.go:121] Processing Rules for Ingress: two/twoapi-syst
I0814 14:26:31.148243       1 routing_rules.go:31] [brownfield] Rule to Targets map: map[rr-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:} {Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**}]]
I0814 14:26:31.148799       1 targets.go:49] [brownfield] Target {} is not blacklisted
I0814 14:26:31.148836       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.148845       1 routing_rules.go:39] [brownfield] Routing Rule rr-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.148855       1 listeners.go:32] [brownfield] Listener fl-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.148863       1 listeners.go:32] [brownfield] Listener listener_dev_agic_https is blacklisted
I0814 14:26:31.148871       1 listeners.go:32] [brownfield] Listener listener_syst_agic_https is blacklisted
I0814 14:26:31.148883       1 listeners.go:103] [brownfield] Listeners AGIC created: fl-e1903c8aa3446b7b3207aec6d6ecba8a
I0814 14:26:31.148897       1 listeners.go:104] [brownfield] Existing Blacklisted Listeners AGIC will retain: fl-e1903c8aa3446b7b3207aec6d6ecba8a, listener_dev_agic_https, listener_syst_agic_https
I0814 14:26:31.148904       1 listeners.go:105] [brownfield] Existing Listeners AGIC will remove: n/a
I0814 14:26:31.148941       1 routing_rules.go:31] [brownfield] Rule to Targets map: map[rr-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:} {Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**}]]
I0814 14:26:31.148980       1 targets.go:49] [brownfield] Target {} is not blacklisted
I0814 14:26:31.149002       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.149021       1 routing_rules.go:39] [brownfield] Routing Rule rr-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.149040       1 pathmaps.go:29] [brownfield] PathMap to Targets map: map[url-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**}]]
I0814 14:26:31.149082       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.149100       1 pathmaps.go:37] [brownfield] Routing PathMap url-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.149112       1 redirects.go:52] [brownfield] Redirects AGIC created: n/a
I0814 14:26:31.149120       1 redirects.go:53] [brownfield] Existing Blacklisted Redirects AGIC will retain: n/a
I0814 14:26:31.149127       1 redirects.go:54] [brownfield] Existing Redirects AGIC will remove: n/a
I0814 14:26:31.149787       1 requestroutingrules.go:349] Attached pool pool-one-oneapi-syst-8080-bp-8080 and http setting bp-one-oneapi-syst-8080-8080-oneapi-syst to path rule: pr-one-oneapi-syst-0
I0814 14:26:31.149890       1 requestroutingrules.go:349] Attached pool pool-two-twoapi-syst-80-bp-8080 and http setting bp-two-twoapi-syst-80-8080-twoapi-syst to path rule: pr-two-twoapi-syst-0
I0814 14:26:31.149943       1 requestroutingrules.go:107] Bound path-based rule: rr-e1903c8aa3446b7b3207aec6d6ecba8a to listener: fl-e1903c8aa3446b7b3207aec6d6ecba8a ([    ], 80) and url path map url-e1903c8aa3446b7b3207aec6d6ecba8a
I0814 14:26:31.149973       1 pathmaps.go:29] [brownfield] PathMap to Targets map: map[url-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**}]]
I0814 14:26:31.150309       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.150329       1 pathmaps.go:37] [brownfield] Routing PathMap url-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.150341       1 pathmaps.go:93] [brownfield] PathMaps AGIC created: url-e1903c8aa3446b7b3207aec6d6ecba8a
I0814 14:26:31.150349       1 pathmaps.go:94] [brownfield] Existing Blacklisted PathMaps AGIC will retain: url-e1903c8aa3446b7b3207aec6d6ecba8a
I0814 14:26:31.150355       1 pathmaps.go:95] [brownfield] Existing PathMaps AGIC will remove: n/a
I0814 14:26:31.150363       1 pathmaps.go:66] [brownfield] Adding urlpath url-e1903c8aa3446b7b3207aec6d6ecba8a to the uniq map
I0814 14:26:31.150370       1 pathmaps.go:63] [brownfield] Merging urlpath url-e1903c8aa3446b7b3207aec6d6ecba8a in existing blacklist and AGIC list
I0814 14:26:31.150393       1 routing_rules.go:31] [brownfield] Rule to Targets map: map[rr-e1903c8aa3446b7b3207aec6d6ecba8a:[{Hostname: Path:} {Hostname: Path:/oneapi_dev/**} {Hostname: Path:/twoapi_dev/**} {Hostname: Path:/oneapi_syst/**} {Hostname: Path:/twoapi_syst/**}]]
I0814 14:26:31.150425       1 targets.go:49] [brownfield] Target {} is not blacklisted
I0814 14:26:31.150461       1 targets.go:45] [brownfield] Target {"Path":"/oneapi_dev/**"} is blacklisted
I0814 14:26:31.150470       1 routing_rules.go:39] [brownfield] Routing Rule rr-e1903c8aa3446b7b3207aec6d6ecba8a is blacklisted
I0814 14:26:31.150482       1 routing_rules.go:95] [brownfield] Rules AGIC created: rr-e1903c8aa3446b7b3207aec6d6ecba8a
I0814 14:26:31.150493       1 routing_rules.go:96] [brownfield] Existing Blacklisted Rules AGIC will retain: rr-e1903c8aa3446b7b3207aec6d6ecba8a
I0814 14:26:31.150499       1 routing_rules.go:97] [brownfield] Existing Rules AGIC will remove: n/a
I0814 14:26:31.150508       1 routing_rules.go:109] [brownfield] Merging path based rule rr-e1903c8aa3446b7b3207aec6d6ecba8a with rule rr-e1903c8aa3446b7b3207aec6d6ecba8a
I0814 14:26:31.150519       1 routing_rules.go:123] [brownfield] Merging path map url-e1903c8aa3446b7b3207aec6d6ecba8a with path map url-e1903c8aa3446b7b3207aec6d6ecba8a
I0814 14:26:31.150529       1 routing_rules.go:127] [brownfield] Deleting path map url-e1903c8aa3446b7b3207aec6d6ecba8a
I0814 14:26:31.150546       1 pathmaps.go:140] [brownfield] Deleting /subscriptions/<<>>/resourceGroups/<<>>/providers/Microsoft.Network/applicationGateways/<<>>/urlPathMaps/url-e1903c8aa3446b7b3207aec6d6ecba8a
I0814 14:26:31.162424       1 mutate_app_gateway.go:174] Generated config:
--
Gateway Config
--
**E0814 14:26:32.058237       1 worker.go:62] Error processing event.network.ApplicationGatewaysClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="InvalidResourceReference" Message="Resource /subscriptions/<<>>/resourceGroups/<<>>/providers/Microsoft.Network/applicationGateways/<<>>/urlPathMaps/url-e1903c8aa3446b7b3207aec6d6ecba8a referenced by resource /subscriptions/<<>>/resourceGroups/<<>>/providers/Microsoft.Network/applicationGateways/<<>>/requestRoutingRules/rr-e1903c8aa3446b7b3207aec6d6ecba8a was not found. Please make sure that the referenced resource exists, and that both resources are in the same region." Details=[]**

Are we doing anything wrong here? Context roots are different in each environment. they are prefixed by appropriate env (_dev/ _syst)

**Comment by 3quanfeng** [comment]
Hi @sudheeranguluri , when you run 
"kubectl get AzureIngressProhibitedTargets -n your_nm"
How many prohibited policies are u able to get? If you find one named "prohibit-all-targets", make sure delete it and try one more time.

**Comment by sudheeranguluri** [comment]
Hello @3quanfeng,

Thanks for response. I was able to find the issue and fixed it this morning. It has to do with how I specified prohibited routes and hostnames in deployment templates. It is fixed now.

Not sure if I can ask different question here or should I create a new issue.

I know AGIC is using hostnames property in order to specify hostnames for gateway, When can we see that property in portal. Based on what I read on internet, it looks like some of them are already seeing it on web portal, But not us. 

Routing is good and no issue with deployments. Just the UI issue on portal.
